xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/WcagContext.java

```diff
@@ -0,0 +1,412 @@
+        /**
+         * @param testMethodName in which the validation happened.
+         * @param url from which these results have been generated
+         * @param pageClassName class of the page validated
+         * @param axeResults results of the validation
+         */
+        private WCAGTestResults(String testMethodName, String url, String pageClassName, Results axeResults)
+        {
+            // Generate the report as soon as the results are in.
+            List<Rule> failingViolations = axeResults.getViolations()
+                .stream()
+                .filter(rule -> !FAILS_ON_RULE.containsKey(rule.getId()) || FAILS_ON_RULE.get(rule.getId()))
+                /* If the ruleid is not defined in FAILS_ON_RULE,
+                the default behavior will be to add it to the fails.
+                In order to resolve these test-suite fails quickly, set them as "false" in FAILS_ON_RULE.
+                 */
+                .collect(Collectors.toList());
+            if (!failingViolations.isEmpty())
+            {
+                this.failReport = XWikiCustomAxeReporter.getReadableAxeResults(testMethodName, pageClassName,
+                        url, failingViolations);
+            }
+
+            List<Rule> warningViolations = axeResults.getViolations()
+                    .stream()
+                    .filter(rule -> FAILS_ON_RULE.containsKey(rule.getId()) && !FAILS_ON_RULE.get(rule.getId()))
+                    .collect(Collectors.toList());
+            if (!warningViolations.isEmpty())
+            {
+                this.warnReport = XWikiCustomAxeReporter.getReadableAxeResults(testMethodName, pageClassName,
+                        url, warningViolations);
+            }
+        }
+        String getFailReport()
+        {
+            return this.failReport;
+        }
+        String getWarnReport()
+        {
+            return this.warnReport;
+        }
+
+
+        boolean isFailEmpty()
+        {
+            return this.failReport.equals("");
+        }
+
+        boolean isWarnEmpty()
+        {
+            return this.warnReport.equals("");
+        }
+    }
+
+    private List<WCAGTestResults> wcagResults = new ArrayList<>();
+
+    private boolean wcagEnabled = true;
+    private long wcagTimer;
+    private final Map<String, ArrayList<String> > wcagValidationCache = new HashMap<>();
+    private String testClassName;
+    private String testMethodName;
+
+    /**
+     * @return the first non-empty fail report from all the validations.
+     */
+    public String getFirstWcagFail()
+    {
+        String report = "";
+        for (WCAGTestResults wcagResult : wcagResults)
+        {
+            if (!wcagResult.isFailEmpty())
+            {
+                report = wcagResult.getFailReport();
+                break;
+            }
+        }
+        return report;
+    }
+
+    /**
+     * Sets the current test class name.
+     * @param testClassName new value to be set.
```

Comment:
missing `the`. See code style at https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/JavaCodeStyle/#HJavadocBestPractices